### PR TITLE
let store access torrent/file events and figure out paths itself

### PR DESCRIPTION
**Pull request by ftreesmilo**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Pass the Torrent and TorrentFile instances to the store to allow stores to make better decisions on how to eliminate file collisions.

**Which issue (if any) does this pull request address?**

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

**Comment by jhiesey** [comment]
@ftreesmilo sorry I've been a bit busy. Still planning to look at this this week.

**Review by jhiesey — CHANGES_REQUESTED** [review]
In addition to having the issues I note here https://github.com/webtorrent/webtorrent/pull/2121#discussion_r671794675

changing the meaning of `file.path` is a breaking change; we'd have to make that webtorrent 2.0.

I think the best solution is what I propose here: https://github.com/webtorrent/webtorrent/pull/2121#discussion_r671794675

File: docs/api.md
Comment (ftreesmilo) [reply docs/api.md:558]: Can you expand on this?  
What is wormhole doing where this would break?

The file.path really doesn't seem to be a safe indicator of a file location, as the torrent and files are backed by chunk-stores.  In a we (and I'd argue even in a native install) getting a file stream through the apis is a safer way to refer to files than getting the file object and trying to use the path.

To me, this information seems more for display purposes than actually useful.  The store is the one who should decide where the file actually resides.

File: lib/torrent.js
Comment (ftreesmilo) [reply lib/torrent.js:483]: That breaks compat with a lot more than just fs-chunk-store.

This could have implications for any store who used this property as a means to identify a file.   If you change this, metadata stores may have used to keep track of files will now be incorrect.  This could lead to orphaned files, cause new files to download when existing data already existed, or data to be lost until the stores are updated...  

I think change is far worse than the other.

File: docs/api.md
Comment (ftreesmilo) [reply docs/api.md:558]: Is the wormhole code open source?   Can I see how file.path is being used there?

**Comment by ftreesmilo** [comment]
yes this is still relevant.   waiting for code review.

**Pull request closed without merging**
